Added committee_session_id to data entries and results

Lionqueen94 opened the pull request:
- Made polling station id and committee session id unique on data entry and results
- Use those for create, update, save and finalise
- Lookup current committee session in each endpoint via a helper function

<!--
- What's the scope of the changes?
- What did you test? Which edge cases did you explicitly take into account?
- Are there things you want reviewers to definitely take a look at?
- Are there specific people you want feedback from?
- Do you have tips on how to test? (For example: data setup, triggering specific errors, etc.)
-->

bramtweedegolf reviewed (dismissed):
Looks good!

praseodym commented on backend/src/data_entry/repository.rs:
```diff
@@ -20,27 +20,34 @@ impl PollingStationDataEntries {
     pub async fn get_row(
         &self,
         polling_station_id: u32,
-    ) -> Result<PollingStationDataEntry, sqlx::Error> {
```
Please keep `sqlx::Error` just like in https://github.com/kiesraad/abacus/pull/1622#discussion_r2159166570

praseodym commented on backend/migrations/3_polling_station_data_entries.sql:
```diff
@@ -1,8 +1,11 @@
-    FOREIGN KEY (polling_station_id) REFERENCES polling_stations (id)
+    FOREIGN KEY (polling_station_id) REFERENCES polling_stations (id),
+    FOREIGN KEY (committee_session_id) REFERENCES committee_sessions (id),
+    CONSTRAINT number UNIQUE (polling_station_id, committee_session_id)
```
Please use a more descriptive name for the constraint

praseodym commented on backend/migrations/4_polling_station_results.sql:
```diff
@@ -1,8 +1,11 @@
-    FOREIGN KEY (polling_station_id) REFERENCES polling_stations (id)
+    FOREIGN KEY (polling_station_id) REFERENCES polling_stations (id),
+    FOREIGN KEY (committee_session_id) REFERENCES committee_sessions (id),
+    CONSTRAINT number UNIQUE (polling_station_id, committee_session_id)
```
Same here, more descriptive name for the constraint

praseodym reviewed (commented):
See comments. And a test suggestion: validate that the uniqueness constraints work as expected.

Lionqueen94 commented:
> See comments. And a test suggestion: validate that the uniqueness constraints work as expected.

I added an extra test, not sure if this tests this enough, could you have a look if this is the case?

praseodym commented:
> > See comments. And a test suggestion: validate that the uniqueness constraints work as expected.
> 
> I added an extra test, not sure if this tests this enough, could you have a look if this is the case?

I'd expect a test like:
- In the first committee session, claim data entry
- Create a second committee session
- Claim data entry again, expect two rows in database

Lionqueen94 commented:
> > > See comments. And a test suggestion: validate that the uniqueness constraints work as expected.
> > 
> > 
> > I added an extra test, not sure if this tests this enough, could you have a look if this is the case?
> 
> I'd expect a test like:
> 
>     * In the first committee session, claim data entry
> 
>     * Create a second committee session
> 
>     * Claim data entry again, expect two rows in database

Changed the test in d899fc5c50100b6b748e68596751b7a99084f588 :)

praseodym commented:
@Lionqueen94 there is a small merge conflict in this branch
